source/common/version/api_version.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "common/version/api_version_struct.h"
+
+// Defines the ApiVersion current version (Envoy::api_version), and oldest
+// version (Envoy::oldest_api_version).
+#include "common/version/api_version_number.h"
+
+namespace Envoy {
+
+/**
+ * Wraps compiled in api versioning.
+ */
+class ApiVersionInfo {
+public:
+  // Returns the most recent API version that is supported by the client.
+  static constexpr ApiVersion const& apiVersion() { return api_version; }
```

Comment:
nit: you don't need const& anymore, constexpr is resolved at compile time anyway

Reply:
Thanks!
I had to remove the reference when removing the `const`. I'm assuming that if the compiler inlines this, then there shouldn't be a copy.